### DISCUSSION TITLE
Custom log levels

### DISCUSSION THREAD

**Discussion by abdosaeedelhassan** [opcodesio/log-viewer#61] [Ideas]
Hi, firstly thanks for log-viewer package.
i want to suggest ability to add custom logs, the reason of it, sometime we need to register logs manually for example errors form apis that comes from mobile applications, this allow to collect logs in one place.
is there any plan to adding this future.

Thanks

**Comment by arukompas** [comment]
Hey @abdosaeedelhassan ,

Do you mean different log **formats** or different log **levels** (like `debug`, `info`, etc)?

At the moment it only supports the default Laravel log format, but in the future we will have the ability to add custom log format drivers so that you can see other types of logs, like Apache, Nginx, MySQL, cron tasks, etc.

**Reply by abdosaeedelhassan** [reply]
I mean different log levels ex clientApp, emaxinerApp, smsLogs, i can work to add this feature if will be accepted.
Thank you

**Reply by arukompas** [reply]
@abdosaeedelhassan I still don't understand where in the log context this would appear 😄 

if you're talking about putting these into different **folders** like `storage/logs/clientApp` and `storage/logs/smsLogs` then that's already supported, just add those paths to the configuration.

Otherwise, can you share an example log contents that you're talking about?

**Comment by abdosaeedelhassan** [comment]
My suggestion as follows:

1- let the log levels to be come from configuration file,
2- adding logWriter class to register the errors in current log file,ex writeFun(level,context)

so that we can append custom log levels and used it in log writer to write the errors inside the file, then will shown in log-viewer page.

**Reply by arukompas** [reply]
The current log levels (`debug`, `info`, `warning`, `error`, etc) are standard in all programming languages and tools. We should stick to this standard and use these default levels because they signify the **severity** of the log, not the contents/context of them.

So perhaps what you're looking for is [custom channels](https://laravel.com/docs/9.x/logging#writing-to-specific-channels).

This way, you can use the log levels to tell the **severity** of the log, while still logging them into separate files for convenience:
```php
Log::channel('smsLogs')->info('SMS sent successfully.');
//
Log::channel('smsLogs')->error('Failed to send SMS');
```

If you go this way, then this package already supports this.

**Reply by abdosaeedelhassan** [reply]
Yes, you are right about standard levels and log channel.
i need to append new levels to the standard (i will not touch the standard levels), only i will add array in config that contains new levels (id,name) that will be append to the standard levels in shown the errors in the log page.
i need this feature because i have api for mobiles apps, and want to register the errors that happen in apps, so that can be easier detected an solved the mobile app develpoer.
